app/src/main/java/com/duckduckgo/app/brokensite/BrokenSiteViewModel.kt

```diff
@@ -16,79 +16,107 @@
+        this.url = url
+        this.blockedTrackers = blockedTrackers
+        this.upgradedHttps = upgradedHttps
+        this.surrogates = surrogates
+    }
 
-        if (viewValue.url.isNullOrBlank()) {
-            command.value = Command.FocusUrl
-        } else {
-            command.value = Command.FocusMessage
-        }
+    fun onCategoryIndexChanged(newIndex: Int) {
+        indexSelected = newIndex
     }
 
-    fun onBrokenSiteUrlChanged(newUrl: String?) {
-        viewState.value = viewState.value?.copy(
-            url = newUrl,
-            submitAllowed = canSubmit(newUrl, viewValue.message)
-        )
+    fun onCategorySelectionCancelled() {
+        indexSelected = viewState.value?.indexSelected ?: -1
     }
 
-    fun onFeedbackMessageChanged(newMessage: String?) {
+    fun onCategoryAccepted() {
         viewState.value = viewState.value?.copy(
-            message = newMessage,
-            submitAllowed = canSubmit(viewValue.url, newMessage)
+            indexSelected = indexSelected,
+            categorySelected = categories.elementAtOrNull(indexSelected),
+            submitAllowed = canSubmit()
         )
     }
 
-    private fun canSubmit(url: String?, feedbackMessage: String?): Boolean {
-
-        if (feedbackMessage.isNullOrBlank()) {
-            return false
-        }
-
-        if (url.isNullOrBlank()) {
-            return false
+    fun onSubmitPressed(webViewVersion: String) {
+        url?.let {
+            val brokenSite = getBrokenSite(it, webViewVersion)
+            brokenSiteSender.submitBrokenSiteFeedback(brokenSite)
+            pixel.fire(Pixel.PixelName.BROKEN_SITE_REPORTED, mapOf(Pixel.PixelParameter.URL to brokenSite.siteUrl))
         }
+        command.value = Command.ConfirmAndFinish
+    }
 
-        return true
+    @VisibleForTesting
+    fun getBrokenSite(url: String, webViewVersion: String): BrokenSite {
+        val category = categories[viewValue.indexSelected]
+        val absoluteUrl = Uri.parse(url).absoluteString
+        return BrokenSite(
+            category = category.key,
+            siteUrl = absoluteUrl,
+            upgradeHttps = upgradedHttps,
+            blockedTrackers = blockedTrackers,
+            surrogates = surrogates,
+            webViewVersion = webViewVersion,
+            siteType = if (Uri.parse(url).isMobileSite) MOBILE else DESKTOP
+        )
     }
 
-    fun onSubmitPressed() {
-        val message = viewValue.message ?: return
-        val url = viewValue.url ?: return
+    private fun canSubmit(): Boolean = categories.elementAtOrNull(indexSelected) != null
 
-        brokenSiteSender.submitBrokenSiteFeedback(message, url)
-        pixel.fire(Pixel.PixelName.BROKEN_SITE_REPORTED, mapOf(Pixel.PixelParameter.URL to url))
-        command.value = Command.ConfirmAndFinish
+    companion object {
+        const val UNKNOWN_VERSION = "unknown"
+        const val MOBILE = "mobile"
+        const val DESKTOP = "desktop"
```

Comment:
can we add more context to the purpose of these constants? can we rename this to make it obviously about the `WebView` version number? etc...

Reply:
Done!